modules/order/src/Form/OrderReassignForm.php

```diff
@@ -96,9 +97,11 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
     $this->submitCustomerForm($form, $form_state);
 
     $values = $form_state->getValues();
-    $this->order->setEmail($values['mail']);
-    $this->order->setCustomerId($values['uid']);
-    $this->order->save();
+
+    /** @var \Drupal\commerce_order\OrderAssignment $assignment_service */
+    $assignment_service = \Drupal::service('commerce_order.order_assignment');
+    $assignment_service->assign($this->order, User::load($values['uid']), TRUE);
```

Comment:
Not sure if the approach is just to get it up and tests run for quick and dirty, but we should inject the service + user storage